app/services/factory.py

```
@@ -0,0 +1,66 @@
+
+def discover_providers():
+    """Automatically discover all ASR providers in the providers directory."""
+    import app.services.providers as providers
+
+    for _, module_name, _ in pkgutil.iter_modules(providers.__path__):
+        if module_name == "base":
+            continue
+
+        try:
+            module = importlib.import_module(f"app.services.providers.{module_name}")
+            for name, obj in inspect.getmembers(module, inspect.isclass):
```

Ruff (0.15.12)

[warning] Loop control variable `name` not used within loop body

Rename unused `name` to `_name`

(B007)